docs/DeveloperGuide.md

```diff
@@ -1,34 +1,579 @@
+
+Given below is an example usage scenario and how the viewing task mechanism behaves at each step.
+
+**Step 1.** The user views a task from the application, by inputting `view 1`. 
+
+**Step 2.** The input will be read in by the `Athena` class. The input will be passed into `LogicManager` where `Parser#parse` will parse the user input to get the command type and details then call`Parse#parseViewCommand`to create a `ViewCommand` object. The `ViewCommand` object is returned to the `LogicManager`.
+
+**Step 3.** The `LogicManager` calls `ViewCommand#execute`. The `TaskList` now retrieve details of 1 task (Assignment 1). The message to show if the task details is subsequently outputted by the `AthenaUi` class to the user.
+
+**Step 4.** Upon completion of execution, `LogicManager` returns a boolean value `false` to `Athena` to allow the continuous run of the program. 
+
+The following sequence diagram illustrates how Step 3 of the viewing task operation works:
+
+![ViewTaskSequenceDiagram](sequenceDiagrams/ViewCommand%20diagram.png)
+
+### List feature
+The list command is facilitated by `LogicManager`, using `Parser` which extracts the command type and parameters from the user input, and stores it in a `ListCommand` that contains the parameters provided by the user. `ListCommand` creates a `Timetable` with the existing `TaskList` to generate the output message.
+
+`Parser`, `ListCommand` and `Timetable` implements the following operations:
+
+* `Parser#parse` - Parses user input to extract the command parameters.
+* `Parser#parseListCommand` - Parses user input to retrieve the filters for the `TaskList` and creates a `ListCommand`.
+* `ListCommand#execute` - Prints the user's tasks using `Timetable#toString`.
+* `Timetable#toString` - Gets a string representation of the user's tasks.
+
+Given below is an example usage scenario and how the *list* command behaves at each step.
+
+**Step 1.** The user requests the application to print a list of the existing tasks, by inputting `list`. 
+
+**Step 2.** The input will be read in by the `Athena` class. Then, the input will be passed into `LogicManager` where `Parser#parse` will be called to parse the user input, to create a `ListCommand` containing the parameters provided by the user. The `ListCommand` object is returned to the `LogicManager`.
+
+**Step 3.** `LogicManager` calls `ListCommand#execute`, which creates a `Timetable` based on the `TaskList`, and the `importance` and `forecast` parameters provided by the user. The `Timetable` returns a string representation of the tasks by calling `Timetable#toString` that is printed to the user.
+
+**Step 4.** Upon completion of execution, `LogicManager` returns a boolean value `false` to `Athena` to indicate that the user has not requested to exit the application. 
+
+The following sequence diagram illustrates how Step 3 of the viewing task operation works:
+
+![ViewTaskSequenceDiagram](sequenceDiagrams/ListCommand%20diagram.png)
```

Comment:
The use of instance destruction (cross symbol) seems inconsistent in the diagram (the first two destruct right after the activation bar, whereas the last one is not). It would look better if they are consistent and given a little space after the activation bar in my opinion.